vector-stores/spring-ai-gemfire/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java

```diff
@@ -0,0 +1,505 @@
+				this.key = key;
+				this.vector = vector;
+				this.metadata = new HashMap<>(metadata);
+				this.metadata.put(contentName, content);
+			}
+
+			public String getKey() {
+				return key;
+			}
+
+			public List<Float> getVector() {
+				return vector;
+			}
+
+			public Map<String, Object> getMetadata() {
+				return metadata;
+			}
+
+		}
+
+	}
+
+	private static final class QueryRequest {
+
+		@JsonProperty("vector")
+		@NonNull
+		private final List<Float> vector;
+
+		@JsonProperty("top-k")
+		private final int k;
+
+		@JsonProperty("k-per-bucket")
+		private final int kPerBucket;
+
+		@JsonProperty("include-metadata")
+		private final boolean includeMetadata;
+
+		public QueryRequest(List<Float> vector, int k, int kPerBucket, boolean includeMetadata) {
+			this.vector = vector;
+			this.k = k;
+			this.kPerBucket = kPerBucket;
+			this.includeMetadata = includeMetadata;
+		}
+
+		public List<Float> getVector() {
+			return vector;
+		}
+
+		public int getK() {
+			return k;
+		}
+
+		public int getkPerBucket() {
+			return kPerBucket;
+		}
+
+		public boolean isIncludeMetadata() {
+			return includeMetadata;
+		}
+
+	}
+
+	private static final class QueryResponse {
+
+		private String key;
+
+		private float score;
+
+		private Map<String, Object> metadata;
+
+		private String getContent(String field) {
+			return (String) metadata.get(field);
+		}
+
+		public void setKey(String key) {
+			this.key = key;
+		}
+
+		public void setScore(float score) {
+			this.score = score;
+		}
+
+		public void setMetadata(Map<String, Object> metadata) {
+			this.metadata = metadata;
+		}
+
+	}
+
+	private static class DeleteRequest {
+
+		@JsonProperty("delete-data")
+		private boolean deleteData = true;
+
+		public DeleteRequest() {
+		}
+
+		public DeleteRequest(boolean deleteData) {
+			this.deleteData = deleteData;
+		}
+
+		public boolean isDeleteData() {
+			return deleteData;
+		}
+
+		public void setDeleteData(boolean deleteData) {
+			this.deleteData = deleteData;
+		}
+
+	}
+
+	@Override
+	public void add(List<Document> documents) {
+		UploadRequest upload = new UploadRequest(documents.stream().map(document -> {
+			// Compute and assign an embedding to the document.
+			document.setEmbedding(this.embeddingClient.embed(document));
+			List<Float> floatVector = document.getEmbedding().stream().map(Double::floatValue).toList();
+			return new UploadRequest.Embedding(document.getId(), floatVector, documentField, document.getContent(),
+					document.getMetadata());
+		}).toList());
+
+		ObjectMapper objectMapper = new ObjectMapper();
+		String embeddingsJson = null;
+		try {
+			String embeddingString = objectMapper.writeValueAsString(upload);
+			embeddingsJson = embeddingString.substring("{\"embeddings\":".length());
+		}
+		catch (JsonProcessingException e) {
+			throw new RuntimeException("Embedding json parsing error :" + e.getMessage());
+		}
+
+		client.post()
+			.uri("/" + INDEX_NAME + EMBEDDINGS)
```

Comment:
INDEX_NAME is not configurable from the rest api calls. This should probably take any index name instead of the final string (constant) one?

The question is also for the remaining api calls that use INDEX_NAME instead of a configured index name.